src/Microsoft.AspNet.Http/Features/FeatureHelpers.cs

```diff
@@ -20,6 +20,7 @@ internal static class FeatureHelpers
             {
                 obj = features.Get<T>();
                 cachedObject = obj;
+                cache.SetFeaturesRevision();
```

Comment:
I don't think you need SetFeaturesRevision, it looks like it's always called after CheckFeatureRevision which already updates _cachedFeaturesRevision.